src/test/objectstore/store_test.cc

```diff
@@ -7159,6 +7163,8 @@ TEST_P(StoreTestSpecificAUSize, BlobReuseOnOverwrite) {
     ASSERT_TRUE(bl_eq(expected, bl));
     ASSERT_EQ(logger->get(l_bluestore_blobs), 1u);
     ASSERT_EQ(logger->get(l_bluestore_extents), 2u);
+    ASSERT_EQ(mempool::bluestore_blob::allocated_items(), 1u);
```

Comment:
@ideepika - do you have any ideas how this modification could fix the following issue @kamoltat shared before:
g057e8043/rpm/el8/BUILD/ceph-18.0.0-88-g057e8043/src/test/objectstore/store_test.cc:7165: Failure
2022-09-28T06:47:07.984 INFO:teuthology.orchestra.run.smithi050.stdout:Expected equality of these values:
2022-09-28T06:47:07.984 INFO:teuthology.orchestra.run.smithi050.stdout:  mempool::bluestore_blob::allocated_items()
2022-09-28T06:47:07.984 INFO:teuthology.orchestra.run.smithi050.stdout:    Which is: 9
2022-09-28T06:47:07.985 INFO:teuthology.orchestra.run.smithi050.stdout:  1u
2022-09-28T06:47:07.985 INFO:teuthology.orchestra.run.smithi050.stdout:    Which is: 1
2022-09-28T06:47:08.498 INFO:teuthology.orchestra.run.smithi050.stdout:==> rm -r bluestore.test_temp_dir

To me this rather looks like some volatile behavior which causes mempool stats to vary. Wouldn't be more reliable to get rid of these stats instead?

Reply:
@ideepika - not sure I understand your point/question :(   The above error was referring to bluestore_blob mempool. onode_cache_shard looks irrelevant in this case...
Anyway my suggestion is just not introduce these two new assertions for bluestore_blob and 
bluestore_extent mempools. And hence avoid this unclear failure which we don't know root cause for yet.

Reply:
sure, updated the changes